movement/watch_faces/complication/toss_up_face.c

```diff
@@ -0,0 +1,793 @@
+            }
+            break;
+        default:
+            break;
+    }
+    watch_display_string(buf, 0);
+}
+
+/** @brief divination method to derive a bit from 32 TRNG bits
+ */
+uint8_t divine_bit(void) {
+    uint32_t stalks;
+    do { // modulo bias filter
+        stalks = get_true_entropy(); // get 32 TRNG bits as stalks
+    } while (stalks >= INT32_MAX || stalks <= 0);
+
+    uint8_t pile1_xor = 0;
+    uint8_t pile2_xor = 0;
+    // Divide the stalks into two piles, alternating ends
+    for (uint8_t i = 0; i < 16; i++) {
+        uint8_t left_bit = (stalks >> (31 - 2*i)) & 1;
+        uint8_t right_bit = (stalks >> (30 - 2*i)) & 1;
+        if (i % 2 == 0) {
+            pile1_xor ^= left_bit;
+            pile2_xor ^= right_bit;
+        } else {
+            pile1_xor ^= right_bit;
+            pile2_xor ^= left_bit;
+        }
+    }
+    // Take the XOR of the pile results
+    uint8_t result_xor = pile1_xor ^ pile2_xor;
+    // Output 1 if result_xor is 1, 0 otherwise
+    return result_xor;
+}
+
+/** @brief get 32 True Random Number bits
+ */
+uint32_t get_true_entropy(void) {
+    #if __EMSCRIPTEN__
+    return rand() % INT32_MAX;
+    #else
+    hri_mclk_set_APBCMASK_TRNG_bit(MCLK);
+    hri_trng_set_CTRLA_ENABLE_bit(TRNG);
+
+    while (!hri_trng_get_INTFLAG_reg(TRNG, TRNG_INTFLAG_DATARDY)); // Wait for TRNG data to be ready
+
+    hri_trng_clear_CTRLA_ENABLE_bit(TRNG);
+    hri_mclk_clear_APBCMASK_TRNG_bit(MCLK);
+    return hri_trng_read_DATA_reg(TRNG); // Read a single 32-bit word from TRNG and return it
+    #endif
+}
+
+// COIN FUNCTIONS /////////////////////////////////////////////////////////////
+
+/** @brief sort tossed coins into a pile of heads and a pile of tails
+ */
+static void _sort_coins(char* token, uint8_t num_bits, uint8_t bits, char* heads, char* tails) {
+    uint8_t num_ones = 0;
+    for (uint8_t i = 0; i < num_bits; i++) {
+        if ((bits >> i) & 1) {
+            *token++ = *heads;
+            num_ones++;
+        }
+    }
+    if ( num_bits < 6 ) {
+        for (uint8_t i = 0; i < (6 - num_bits); i++) {
+            *token++ = ' ';
+        }
+    }
+    for (uint8_t i = 0; i < (num_bits - num_ones); i++) {
+
+        *token++ = *tails;
+    }
+}
+
+/** @brief convert bool array of coinflips to integer for sorting
+ */
+void _display_coins(char* token, bool* bit_array, uint8_t length, toss_up_state_t *state) {
+    uint8_t bits = 0;
+    for (uint8_t i = 0; i < length; i++) {
+        if (bit_array[i]) {
+            bits |= (1 << (length - 1 - i));
+        }
+    }
+    _sort_coins(token, length, bits, &state->coin_style[0], &state->coin_style[1]);
+}
+
+/** @brief coin animation
+ */
+static void _coin_animation(toss_up_state_t *state) {
+    bool heads = false;
+    bool tails = false;
+    for (uint8_t i = 0; i < state->coin_num; i++) {
+        if (state->coins[i] == true) {
+            heads++;
+        } else {
+            tails++;
+        }
```

Comment:
These lines trigger a warning about incrementing a boolean — is that what you're trying to do here? Just want to clarify because I'd like to address the warning but I'm not sure what the intent was.

Reply:
Oh yikes. Will fix. I counted heads and tailes before and then realized I only need booleans here. Weird that it still compiled and worked. Will fix asap.